app/api/dao/user.py

```diff
@@ -143,27 +147,33 @@ def get_user_by_username(username: str):
         return UserModel.find_by_username(username)
 
     @staticmethod
-    def list_users(user_id: int, search_query: str = "", is_verified = None):
+    def list_users(user_id: int, search_query: str = "", page: int = DEFAULT_PAGE, per_page: int = DEFAULT_USERS_PER_PAGE, is_verified = None):
         """ Retrieves a list of verified users with the specified ID.
         
         Arguments:
             user_id: The ID of the user to be listed.
             search_query: The search query for name of the users to be found.
             is_verified: Status of the user's verification; None when provided as an argument.
+            page: The page of users to be returned
+            per_page: The number of users to return per page
         
         Returns:
             A list of users matching conditions and the HTTP response code.
         
         """
 
-        users_list = UserModel.query.filter(UserModel.id != user_id).all()
+        users_list = UserModel.query.filter(
+            UserModel.id != user_id,
+            not is_verified or UserModel.is_email_verified,
+            func.lower(UserModel.name).contains(search_query.lower())
+        ).order_by(UserModel.id).paginate(page=page,
+                                          per_page=per_page,
+                                          error_out=False,
+                                          max_per_page=UserDAO.MAX_USERS_PER_PAGE).items
+
         list_of_users = [
             user.json()
-            for user in filter(
-                lambda user: (not is_verified or user.is_email_verified)
-                and search_query.lower() in user.name.lower(),
-                users_list,
-            )
+            for user in users_list
         ]
 
```

Comment:
when len of list_of_users is 0 then raise an error with certain message.

Reply:
@satya7289 we decided in a project meeting to leave this specific change for later, since it isn't critical. Here's the comment: https://github.com/anitab-org/mentorship-backend/pull/541#issuecomment-629836970 There is further discussion in this PR on why this is being left for out of this PR.

Reply:
@isabelcosta 
As a solution to this I point out, instead of return the message if raised a Badrequest exception will solve the issue.  [comment](https://github.com/anitab-org/mentorship-backend/pull/541#issuecomment-627031572)

Reply:
If it is handle in different issue then leave this part, it is ok.

Reply:
> @isabelcosta
> As a solution to this I point out, instead of return the message if raised a Badrequest exception will solve the issue. [comment](https://github.com/anitab-org/mentorship-backend/pull/541#issuecomment-627031572)

sounds good :) we can create a follow-up issue to handle this.

Reply:
Ok